### PR TITLE
[new release] mirage-types, mirage-runtime, mirage-types-lwt and mirage (3.10.3)

### PR DESCRIPTION
**Pull request by dinosaure**
Module type definitions for MirageOS applications

- Project page: <a href="https://github.com/mirage/mirage">https://github.com/mirage/mirage</a>

##### CHANGES:

- Adapt to conduit 4.0.0 and cohttp 4.0.0 (@dinosaure mirage/mirage#1221)

### CODE REVIEW

**Comment by hannesm** [comment]
any chance to get this merged? :) thanks a lot. (the CI failures look unrelated, lots of cancellation but no new failure AFAICT)

File: packages/mirage-runtime/mirage-runtime.3.10.3/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
```
Comment (mseri) [inline packages/mirage-runtime/mirage-runtime.3.10.3/opam:21]:
```suggestion
  "dune" {>= "2.7"}
```
According to https://github.com/mirage/mirage/blob/master/dune-project#L1

File: packages/mirage-types-lwt/mirage-types-lwt.3.10.3/opam [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends:   [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
```
Comment (mseri) [inline packages/mirage-types-lwt/mirage-types-lwt.3.10.3/opam:19]:
```suggestion
  "dune" {>= "2.7"}
```
According to https://github.com/mirage/mirage/blob/master/dune-project#L1

File: packages/mirage-types/mirage-types.3.10.3/opam [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
```
Comment (mseri) [inline packages/mirage-types/mirage-types.3.10.3/opam:21]:
```suggestion
  "dune" {>= "2.7"}
```
According to https://github.com/mirage/mirage/blob/master/dune-project#L1

File: packages/mirage/mirage.3.10.3/opam [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
```
Comment (mseri) [inline packages/mirage/mirage.3.10.3/opam:21]:
```suggestion
  "dune" {>= "2.7"}
```
According to https://github.com/mirage/mirage/blob/master/dune-project#L1

**Comment by mseri** [comment]
Yes, it happens on PRs that have lots of revdeps to test. Most of the cancellations are in fact successful builds.

I have another look at the revdeps, looks good. Do you want to change `dune-project` to `(lang dune 2.0)` or it is ok to update the bounds to 2.7?

**Comment by hannesm** [comment]
@mseri thanks for your review. The mirage 3.x is released from the "3" branch in mirage/mirage, not the master branch (which is prepared for MirageOS 4) -- so in the 3 branch (similar to the archives pointed to by this release) the dune-project says "lang dune 2.0" :)

**Comment by mseri** [comment]
oops, my bad, sorry :)

**Pull request merged**